src/internal/operations/restore.go

```diff
@@ -127,75 +145,73 @@ func (op *RestoreOperation) Run(ctx context.Context) (restoreDetails *details.De
 		"backup_id", op.BackupID,
 		"service", op.Selectors.Service)
 
-	deets, err := op.do(ctx)
+	// -----
+	// Execution
+	// -----
+
+	deets, err := op.do(ctx, &opStats, detailsStore, start)
 	if err != nil {
+		// No return here!  We continue down to persistResults, even in case of failure.
 		logger.Ctx(ctx).
 			With("err", err).
-			Errorw("restore operation", clues.InErr(err).Slice()...)
+			Errorw("doing restore", clues.InErr(err).Slice()...)
+		op.Errors.Fail(errors.Wrap(err, "doing restore"))
+		opStats.readErr = op.Errors.Err()
+	}
 
-		return nil, err
+	// -----
+	// Persistence
+	// -----
+
+	err = op.persistResults(ctx, start, &opStats)
```

Comment:
maybe we should rename this function at some point

Reply:
hehe, probably so.  Another day perhaps.